public/app/features/dashboard/components/ShareModal/ShareLink.tsx

```diff
@@ -44,23 +47,33 @@ export class ShareLink extends PureComponent<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props, prevState: State) {
-    const { useCurrentTimeRange, includeTemplateVars, selectedTheme } = this.state;
+    const { useCurrentTimeRange, includeTemplateVars, useShortUrl, selectedTheme } = this.state;
     if (
       prevState.useCurrentTimeRange !== useCurrentTimeRange ||
       prevState.includeTemplateVars !== includeTemplateVars ||
-      prevState.selectedTheme.value !== selectedTheme.value
+      prevState.selectedTheme.value !== selectedTheme.value ||
+      prevState.useShortUrl !== useShortUrl
     ) {
       this.buildUrl();
     }
   }
 
   buildUrl = () => {
     const { panel } = this.props;
-    const { useCurrentTimeRange, includeTemplateVars, selectedTheme } = this.state;
+    const { useCurrentTimeRange, includeTemplateVars, useShortUrl, selectedTheme } = this.state;
 
     const shareUrl = buildShareUrl(useCurrentTimeRange, includeTemplateVars, selectedTheme.value, panel);
     const imageUrl = buildImageUrl(useCurrentTimeRange, includeTemplateVars, selectedTheme.value, panel);
-    this.setState({ shareUrl, imageUrl });
+
+    if (useShortUrl) {
+      getBackendSrv()
+        .post(`/api/goto`, {
+          path: shareUrl.replace(buildHostUrl(), ''),
+        })
+        .then(uid => this.setState({ shareUrl: buildShortUrl(uid), imageUrl }));
```

Comment:
i find simple promises way easier to read/work with than a bunch of redux stuff spread out all over the place, but i recognize im going against convention here. happy to re-learn the redux way to do this if it's essential